src/metabase/server/middleware/session.clj

```diff
@@ -175,37 +179,51 @@
 
 ;; Because this query runs on every single API request it's worth it to optimize it a bit and only compile it to SQL
 ;; once rather than every time
-(def ^:private ^{:arglists '([db-type max-age-minutes session-type])} session-with-id-query
+(def ^:private ^{:arglists '([db-type max-age-minutes session-type enable-advanced-permissions?])} session-with-id-query
   (memoize
-   (fn [db-type max-age-minutes session-type]
+   (fn [db-type max-age-minutes session-type enable-advanced-permissions?]
      (first
       (db/honeysql->sql
-       {:select    [[:session.user_id :metabase-user-id]
-                    [:user.is_superuser :is-superuser?]
-                    [:user.locale :user-locale]]
-        :from      [[Session :session]]
-        :left-join [[User :user] [:= :session.user_id :user.id]]
-        :where     [:and
-                    [:= :user.is_active true]
-                    [:= :session.id (hsql/raw "?")]
-                    (let [oldest-allowed (sql.qp/add-interval-honeysql-form db-type :%now (- max-age-minutes) :minute)]
-                      [:> :session.created_at oldest-allowed])
-                    [:= :session.anti_csrf_token (case session-type
-                                                   :normal         nil
-                                                   :full-app-embed "?")]]
-        :limit     1})))))
+       (cond->
+         {:select    [[:session.user_id :metabase-user-id]
+                      [:user.is_superuser :is-superuser?]
+                      [:user.locale :user-locale]]
+          :from      [[Session :session]]
+          :left-join [[User :user] [:= :session.user_id :user.id]
+                      ]
+          :where     [:and
+                      [:= :user.is_active true]
+                      [:= :session.id (hsql/raw "?")]
+                      (let [oldest-allowed (sql.qp/add-interval-honeysql-form db-type :%now (- max-age-minutes) :minute)]
+                        [:> :session.created_at oldest-allowed])
+                      [:= :session.anti_csrf_token (case session-type
+                                                     :normal         nil
+                                                     :full-app-embed "?")]]
+          :limit     1}
+
+         enable-advanced-permissions?
+         (->
+          (hh/merge-select
+           [:pgm.is_group_manager :is-group-manager?])
+          (hh/merge-left-join
+           [PermissionsGroupMembership :pgm] [:and
+                                              [:= :pgm.user_id :user.id]
+                                              [:is :pgm.is_group_manager true]]))))))))
```

Comment:
@noahmoss I updated this so we only query `is-group-manager?` if `advanced-permissions` is available